% Options for packages loaded elsewhere
\PassOptionsToPackage{unicode}{hyperref}
\PassOptionsToPackage{hyphens}{url}
%
\documentclass[
]{book}
\usepackage{amsmath,amssymb}
\usepackage{lmodern}
\usepackage{ifxetex,ifluatex}
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
  \usepackage{textcomp} % provide euro and other symbols
\else % if luatex or xetex
  \usepackage{unicode-math}
  \defaultfontfeatures{Scale=MatchLowercase}
  \defaultfontfeatures[\rmfamily]{Ligatures=TeX,Scale=1}
\fi
% Use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
\IfFileExists{microtype.sty}{% use microtype if available
  \usepackage[]{microtype}
  \UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}
\makeatletter
\@ifundefined{KOMAClassName}{% if non-KOMA class
  \IfFileExists{parskip.sty}{%
    \usepackage{parskip}
  }{% else
    \setlength{\parindent}{0pt}
    \setlength{\parskip}{6pt plus 2pt minus 1pt}}
}{% if KOMA class
  \KOMAoptions{parskip=half}}
\makeatother
\usepackage{xcolor}
\IfFileExists{xurl.sty}{\usepackage{xurl}}{} % add URL line breaks if available
\IfFileExists{bookmark.sty}{\usepackage{bookmark}}{\usepackage{hyperref}}
\hypersetup{
  pdftitle={ensemblemerge package documentation},
  pdfauthor={Erik Skie},
  hidelinks,
  pdfcreator={LaTeX via pandoc}}
\urlstyle{same} % disable monospaced font for URLs
\usepackage{color}
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
% Add ',fontsize=\small' for more characters per line
\usepackage{framed}
\definecolor{shadecolor}{RGB}{248,248,248}
\newenvironment{Shaded}{\begin{snugshade}}{\end{snugshade}}
\newcommand{\AlertTok}[1]{\textcolor[rgb]{0.94,0.16,0.16}{#1}}
\newcommand{\AnnotationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\AttributeTok}[1]{\textcolor[rgb]{0.77,0.63,0.00}{#1}}
\newcommand{\BaseNTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\BuiltInTok}[1]{#1}
\newcommand{\CharTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\CommentTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\CommentVarTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ConstantTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ControlFlowTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\DataTypeTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{#1}}
\newcommand{\DecValTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\DocumentationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\ErrorTok}[1]{\textcolor[rgb]{0.64,0.00,0.00}{\textbf{#1}}}
\newcommand{\ExtensionTok}[1]{#1}
\newcommand{\FloatTok}[1]{\textcolor[rgb]{0.00,0.00,0.81}{#1}}
\newcommand{\FunctionTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\ImportTok}[1]{#1}
\newcommand{\InformationTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\newcommand{\KeywordTok}[1]{\textcolor[rgb]{0.13,0.29,0.53}{\textbf{#1}}}
\newcommand{\NormalTok}[1]{#1}
\newcommand{\OperatorTok}[1]{\textcolor[rgb]{0.81,0.36,0.00}{\textbf{#1}}}
\newcommand{\OtherTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{#1}}
\newcommand{\PreprocessorTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textit{#1}}}
\newcommand{\RegionMarkerTok}[1]{#1}
\newcommand{\SpecialCharTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\SpecialStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\StringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\VariableTok}[1]{\textcolor[rgb]{0.00,0.00,0.00}{#1}}
\newcommand{\VerbatimStringTok}[1]{\textcolor[rgb]{0.31,0.60,0.02}{#1}}
\newcommand{\WarningTok}[1]{\textcolor[rgb]{0.56,0.35,0.01}{\textbf{\textit{#1}}}}
\usepackage{longtable,booktabs,array}
\usepackage{calc} % for calculating minipage widths
% Correct order of tables after \paragraph or \subparagraph
\usepackage{etoolbox}
\makeatletter
\patchcmd\longtable{\par}{\if@noskipsec\mbox{}\fi\par}{}{}
\makeatother
% Allow footnotes in longtable head/foot
\IfFileExists{footnotehyper.sty}{\usepackage{footnotehyper}}{\usepackage{footnote}}
\makesavenoteenv{longtable}
\usepackage{graphicx}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
% Set default figure placement to htbp
\makeatletter
\def\fps@figure{htbp}
\makeatother
\setlength{\emergencystretch}{3em} % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}
\usepackage{booktabs}
\ifluatex
  \usepackage{selnolig}  % disable illegal ligatures
\fi
\usepackage[]{natbib}
\bibliographystyle{apalike}

\title{ensemblemerge package documentation}
\author{Erik Skie}
\date{2021-03-26}

\begin{document}
\maketitle

{
\setcounter{tocdepth}{1}
\tableofcontents
}
\hypertarget{package-information}{%
\chapter{Package Information}\label{package-information}}

\hypertarget{description}{%
\section{Description}\label{description}}

\textbf{ensemblemerge} is a package that implements a common work flow for several single cell RNA sequence integration methods including Seurat V3, Scanorama, Harmony, Liger, and bbknn. The merging process is designed to be as streamlined as possible for the user, only requiring a \texttt{SummarizedExperiment} or \texttt{SingleCellExperiment} object and a specification of which method should be used to integrate in \texttt{setParams()}.

\hypertarget{quick-start-guide}{%
\subsection{Quick Start Guide}\label{quick-start-guide}}

Once \textbf{ensemblemerge} is installed, merging batches can be performed by the following:

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ensemblemerge)}

\NormalTok{data }\CommentTok{\#some SingleCellExperiment or SummarizedExperiment object with batches to be integrated}

\NormalTok{params }\OtherTok{=} \FunctionTok{setParams}\NormalTok{() }\CommentTok{\#different integration methods can be selected by setting method = c("Seurat", "Harmony", "Liger", "Scanorama", "BBKNN")}

\NormalTok{merged\_data }\OtherTok{=} \FunctionTok{Merge}\NormalTok{(params, data)}
\end{Highlighting}
\end{Shaded}

A \texttt{SingleCellExperiment} object is returned in \texttt{merged\_data} with accompanying dimension reductions if applicable for the integration methods.

\hypertarget{prerequisites}{%
\section{Prerequisites}\label{prerequisites}}

The following R packages are required for installation of \textbf{ensemblemerge}.

\begin{Shaded}
\begin{Highlighting}[]
\ControlFlowTok{if}\NormalTok{ (}\SpecialCharTok{!}\FunctionTok{requireNamespace}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{, }\AttributeTok{quietly =} \ConstantTok{TRUE}\NormalTok{))}
    \FunctionTok{install.packages}\NormalTok{(}\StringTok{"BiocManager"}\NormalTok{)}
\NormalTok{BiocManager}\SpecialCharTok{::}\FunctionTok{install}\NormalTok{(}\FunctionTok{c}\NormalTok{(}\StringTok{"SingleCellExperiment"}\NormalTok{, }\StringTok{"SummarizedExperiment"}\NormalTok{, }\StringTok{"LoomExperiment"}\NormalTok{))}
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"Seurat"}\NormalTok{)}
\NormalTok{devtools}\SpecialCharTok{::}\FunctionTok{install\_github}\NormalTok{(}\StringTok{\textquotesingle{}theislab/kBET\textquotesingle{}}\NormalTok{)}
\NormalTok{devtools}\SpecialCharTok{::}\FunctionTok{install\_github}\NormalTok{(}\StringTok{\textquotesingle{}satijalab/seurat{-}wrappers\textquotesingle{}}\NormalTok{)}
\NormalTok{devtools}\SpecialCharTok{::}\FunctionTok{install\_github}\NormalTok{(}\StringTok{"cellgeni/sceasy"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

If you would like to use the python based integration methods, the following packages must also be installed in your python environment. If you do not have a python environment set, users can also call \texttt{ensemblemerge::install\_python\_packages()} which will install the python prerequisites into a reticulated python environment.

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{pip3 install scanpy}
\NormalTok{pip3 install bbknn}
\NormalTok{pip3 install scanorama}
\end{Highlighting}
\end{Shaded}

The \textbf{ensemblemerge} package can be installed from CRAN or Github (\emph{package is currently in alpha and only available on github to collaborators}):

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{install.packages}\NormalTok{(}\StringTok{"ensemblemerge"}\NormalTok{)}
\CommentTok{\# or the development version}
\CommentTok{\# devtools::install\_github("erikjskie/ensemblemerge")}
\end{Highlighting}
\end{Shaded}

\hypertarget{merge-example-vignettes}{%
\chapter{Merge example vignettes}\label{merge-example-vignettes}}

For more information on specific function parameters and class structures, see sections \ref{functions} and \ref{classes} respectively. This chapter covers basic vignettes of different integration methods using the example dataset provided in our package under \texttt{dendrites} from \href{'https://science.sciencemag.org/content/356/6335/eaah4573.abstract?casa_token=Il6aAbDukAcAAAAA:g6j1TW5RkK2J488DrHHgSmrCRwYTOgFHoH_X6H_bDL7PeuX5h3Ak2zMPEPAi4KKr_WaFZ2ekMc6uwg8'}{Villani et al.~2017}.

\hypertarget{processing-data-without-merging-batches}{%
\section{Processing data without merging batches}\label{processing-data-without-merging-batches}}

The following work flow is for processing data without performing batch correction. This can be done to prepare your data for downstream visualization without having to manually perform normalizing and scaling.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ensemblemerge)}

\NormalTok{params }\OtherTok{=} \FunctionTok{setParams}\NormalTok{(}\AttributeTok{method =} \StringTok{"Uncorrected"}\NormalTok{, }\AttributeTok{batch =} \StringTok{"batchlb"}\NormalTok{) }\CommentTok{\#any additional processing argument can be added here, see chapter 3: classes}

\NormalTok{uncorrected }\OtherTok{=} \FunctionTok{Merge}\NormalTok{(params, dendrites)}
\end{Highlighting}
\end{Shaded}

\texttt{uncorrected} contains normalized and scaled read counts and a PCA reduction. generating a UMAP reduction can be done by the following.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(Seurat)}
\NormalTok{uncorrected }\OtherTok{=} \FunctionTok{as.Seurat}\NormalTok{(uncorrected, }\AttributeTok{counts =} \StringTok{"counts"}\NormalTok{, }\AttributeTok{data =} \StringTok{"logcounts"}\NormalTok{)}
\NormalTok{uncorrected }\OtherTok{\textless{}{-}} \FunctionTok{RunUMAP}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{"PCA"}\NormalTok{, }\AttributeTok{dims =} \DecValTok{1}\SpecialCharTok{:}\DecValTok{20}\NormalTok{, }\AttributeTok{seed.use =} \DecValTok{1}\NormalTok{)}
\NormalTok{uncorrected }\OtherTok{\textless{}{-}} \FunctionTok{FindNeighbors}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{\textquotesingle{}umap\textquotesingle{}}\NormalTok{, }\AttributeTok{dims =} \DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{, }\AttributeTok{verbose =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{uncorrected }\OtherTok{\textless{}{-}} \FunctionTok{FindClusters}\NormalTok{(uncorrected, }\AttributeTok{algorithm =} \DecValTok{3}\NormalTok{, }\AttributeTok{resolution =} \FloatTok{0.01}\NormalTok{, }\AttributeTok{verbose =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{uncorrected}\SpecialCharTok{$}\NormalTok{cluster }\OtherTok{\textless{}{-}} \FunctionTok{Idents}\NormalTok{(uncorrected)}
\end{Highlighting}
\end{Shaded}

Visualize the UMAP.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{Idents}\NormalTok{(uncorrected) }\OtherTok{=} \StringTok{"batchlb"}
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Uncorrected Batch"}\NormalTok{)}
\FunctionTok{Idents}\NormalTok{(uncorrected) }\OtherTok{=} \StringTok{"CellType"}
\NormalTok{p2 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Uncorrected Cell Type"}\NormalTok{)}
\NormalTok{p1 }\SpecialCharTok{+}\NormalTok{ p2}
\end{Highlighting}
\end{Shaded}

\begin{figure}

{\centering \includegraphics[width=33.33in,height=0.8\textheight]{_book/ensemblemerge_files/images/uncorrected_batch} 

}

\caption{Uncorrected UMAP of single cell data by batch and cell type}\label{fig:unnamed-chunk-9}
\end{figure}

\hypertarget{merging-with-seurat}{%
\section{Merging with Seurat}\label{merging-with-seurat}}

This workflow is for integrating batches with the \href{'https://satijalab.org/seurat/articles/integration_introduction.html'}{Seurat V3} batch correction method.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(ensemblemerge)}

\NormalTok{params }\OtherTok{=} \FunctionTok{setParams}\NormalTok{(}\AttributeTok{method =} \StringTok{"Seurat"}\NormalTok{, }\AttributeTok{batch =} \StringTok{"batchlb"}\NormalTok{) }\CommentTok{\#any additional processing argument can be added here, see chapter 3: classes}

\NormalTok{seurat\_integrate }\OtherTok{=} \FunctionTok{Merge}\NormalTok{(params, dendrites)}
\end{Highlighting}
\end{Shaded}

\texttt{seurat\_integrate} contains normalized and scaled read counts and a PCA reduction. generating a UMAP reduction can be done by the following.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(Seurat)}
\NormalTok{seurat\_integrate }\OtherTok{=} \FunctionTok{as.Seurat}\NormalTok{(seurat\_integrate, }\AttributeTok{counts =} \StringTok{"logcounts"}\NormalTok{, }\AttributeTok{data =} \StringTok{"logcounts"}\NormalTok{)}
\NormalTok{seurat\_integrate }\OtherTok{\textless{}{-}} \FunctionTok{RunUMAP}\NormalTok{(seurat\_integrate, }\AttributeTok{reduction =} \StringTok{"PCA"}\NormalTok{, }\AttributeTok{dims =} \DecValTok{1}\SpecialCharTok{:}\DecValTok{20}\NormalTok{, }\AttributeTok{seed.use =} \DecValTok{1}\NormalTok{)}
\NormalTok{seurat\_integrate }\OtherTok{\textless{}{-}} \FunctionTok{FindNeighbors}\NormalTok{(seurat\_integrate, }\AttributeTok{reduction =} \StringTok{\textquotesingle{}umap\textquotesingle{}}\NormalTok{, }\AttributeTok{dims =} \DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{, }\AttributeTok{verbose =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{seurat\_integrate }\OtherTok{\textless{}{-}} \FunctionTok{FindClusters}\NormalTok{(seurat\_integrate, }\AttributeTok{algorithm =} \DecValTok{3}\NormalTok{, }\AttributeTok{resolution =} \FloatTok{0.01}\NormalTok{, }\AttributeTok{verbose =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{seurat\_integrate}\SpecialCharTok{$}\NormalTok{cluster }\OtherTok{\textless{}{-}} \FunctionTok{Idents}\NormalTok{(seurat\_integrate)}
\end{Highlighting}
\end{Shaded}

Visualize the UMAP (\emph{including the uncorrected data for reference}).

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{Idents}\NormalTok{(uncorrected) }\OtherTok{=} \StringTok{"batchlb"}
\NormalTok{p1 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Uncorrected Batch"}\NormalTok{)}
\FunctionTok{Idents}\NormalTok{(uncorrected) }\OtherTok{=} \StringTok{"CellType"}
\NormalTok{p2 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(uncorrected, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Uncorrected Cell Type"}\NormalTok{)}
\FunctionTok{Idents}\NormalTok{(seurat\_integrate) }\OtherTok{=} \StringTok{"batchlb"}
\NormalTok{p3 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(seurat\_integrate, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Corrected Batch"}\NormalTok{)}
\FunctionTok{Idents}\NormalTok{(seurat\_integrate) }\OtherTok{=} \StringTok{"CellType"}
\NormalTok{p4 }\OtherTok{\textless{}{-}} \FunctionTok{DimPlot}\NormalTok{(seurat\_integrate, }\AttributeTok{reduction =} \StringTok{"umap"}\NormalTok{, }\AttributeTok{pt.size =} \FloatTok{0.5}\NormalTok{, }\AttributeTok{shuffle =} \ConstantTok{TRUE}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ ggplot2}\SpecialCharTok{::}\FunctionTok{ggtitle}\NormalTok{(}\AttributeTok{label =} \StringTok{"Corrected Cell Type"}\NormalTok{)}
\NormalTok{p1 }\SpecialCharTok{+}\NormalTok{ p2 }\SpecialCharTok{+}\NormalTok{ p3 }\SpecialCharTok{+}\NormalTok{ p4}
\end{Highlighting}
\end{Shaded}

\begin{figure}

{\centering \includegraphics[width=33.33in,height=0.8\textheight]{_book/ensemblemerge_files/images/seurat_integrate} 

}

\caption{Seurat V3 corrected UMAP of single cell data by batch and cell type}\label{fig:unnamed-chunk-13}
\end{figure}

\hypertarget{functions}{%
\chapter{functions}\label{functions}}

\hypertarget{setparams}{%
\section{setParams()}\label{setparams}}

\texttt{setParams()} generates a \texttt{Params} object of class specified by the \texttt{method} argument, i.e.~\texttt{method\ =\ "Seurat"} returns a \texttt{SeuratParams} object. The slots of the object can be modified on calling \texttt{setParams()} by passing in the name of the slot and the value into the function. i.e \texttt{batch\ =\ "batch"}

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{setParams}\NormalTok{(}\AttributeTok{method =} \StringTok{"Seurat"}\NormalTok{, ...)}
\end{Highlighting}
\end{Shaded}

\textbf{Arguments}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}@{}}
\toprule
Argument & Description \\
\midrule
\endhead
\textbf{method} & character vector specifying the integration method, currently the following methods are available \texttt{c("Seurat",\ "Harmony",\ "Liger",\ "Scanorama",\ "BBKNN")} \\
\textbf{\ldots{}} & additional arguments that modify the slots of the \texttt{Params} object specified by the \texttt{methods} argument \\
\bottomrule
\end{longtable}

\hypertarget{merge}{%
\section{Merge()}\label{merge}}

\texttt{Merge()} generates a \texttt{SingleCellExperiment} object that has been batch integrated by the method specified in the \texttt{params} object.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{Merge}\NormalTok{(params, data)}
\end{Highlighting}
\end{Shaded}

\textbf{Arguments}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}@{}}
\toprule
Argument & Description \\
\midrule
\endhead
\textbf{params} & A \texttt{params} object generated by the \texttt{setParams} function \\
\textbf{data} & A \texttt{SummarizedExperiment} or \texttt{SingleCellExperiment} object with a metadata identifying the batches \\
\bottomrule
\end{longtable}

\hypertarget{classes}{%
\chapter{classes}\label{classes}}

\hypertarget{uncorrectedparams}{%
\section{UncorrectedParams}\label{uncorrectedparams}}

Class generated from \texttt{setParams(method\ =\ "Uncorrected")} for normalizing and scaling data without batch correction.

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{UncorrectedParams}\NormalTok{(}\AttributeTok{batch =} \StringTok{"batch"}\NormalTok{,}
          \AttributeTok{dimreduc\_names =} \FunctionTok{c}\NormalTok{(}\StringTok{"PCA"} \OtherTok{=} \StringTok{"pca"}\NormalTok{,}
                            \StringTok{"UMAP"} \OtherTok{=} \StringTok{"umap"}\NormalTok{,}
                            \StringTok{"tSNE"} \OtherTok{=} \StringTok{"tsne"}\NormalTok{),}
          \AttributeTok{npcs =} \DecValTok{20}\NormalTok{, }
          \AttributeTok{norm\_data =} \ConstantTok{TRUE}\NormalTok{,}
          \AttributeTok{scaling =} \ConstantTok{TRUE}\NormalTok{,}
          \AttributeTok{regressUMI =} \ConstantTok{FALSE}\NormalTok{,}
          \AttributeTok{min\_cells =} \DecValTok{10}\NormalTok{,}
          \AttributeTok{min\_genes =} \DecValTok{300}\NormalTok{,}
          \AttributeTok{norm\_method =} \StringTok{"LogNormalize"}\NormalTok{,}
          \AttributeTok{scale\_factor =} \DecValTok{10000}\NormalTok{,}
          \AttributeTok{numHVG =} \DecValTok{2000}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\textbf{Class Structure}

\begin{longtable}[]{@{}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}
  >{\raggedright\arraybackslash}p{(\columnwidth - 2\tabcolsep) * \real{0.50}}@{}}
\toprule
\textbf{Slot} & \textbf{Description} \\
\midrule
\endhead
\textbf{batch} & metadata name specifying the batches \\
\textbf{dimreduc\_names} & Names of the dimension reduction objects to return \\
\textbf{npcs} & Number of PCA dimensions to return in PCA reduction \\
\textbf{scaling} & boolean specifying whether scaling should be performed \\
\textbf{regressUMI} & boolean specifying whether the number of UMI should be regressed \\
\textbf{norm\_method} & character specifying what normalization method to use \\
\textbf{min\_cells} & minimum number of cells expressing a gene to keep for downstream analysis \\
\textbf{min\_genes} & minimum number of genes a cell must express to keep for downstream analysis \\
\textbf{scale\_factor} & Number to scale gene expression to \\
\textbf{numHVG} & Top N number of genes to assign as highly variable \\
\bottomrule
\end{longtable}

  \bibliography{book.bib,packages.bib}

\end{document}
